packages/cdk/lib/stacks/engines/toil-engine-construct.ts

```diff
@@ -74,17 +75,23 @@ export class ToilEngineConstruct extends EngineConstruct {
     };
   }
 
-  private getEngineServiceDefinition(vpc: IVpc, subnets: SubnetSelection, serviceContainer: ServiceContainer, logGroup: ILogGroup) {
+  private getEngineServiceDefinition(
+    vpc: IVpc,
+    subnets: SubnetSelection,
+    serviceContainer: ServiceContainer,
+    logGroup: ILogGroup,
+    params: ContextAppParameters
+  ) {
     const id = "Engine";
     const definition = new FargateTaskDefinition(this, "EngineTaskDef", {
       taskRole: this.engineRole,
-      cpu: serviceContainer.cpu,
-      memoryLimitMiB: serviceContainer.memoryLimitMiB,
+      cpu: (params.vCpus ? params.vCpus * 1024 : undefined) || serviceContainer.cpu,
```

Comment:
Here we go again... I think if some number has to be repeated more than once, it should be moved to a constant.